plugins/source/aws/client/client.go

```diff
@@ -359,132 +217,33 @@ func Configure(ctx context.Context, logger zerolog.Logger, spec specs.Source, op
+			if err != nil {
+				return err
 			}
-			var ae smithy.APIError
-			if errors.As(err, &ae) {
-				if strings.Contains(ae.ErrorCode(), "AccessDenied") {
-					logger.Warn().Str("account", account.AccountName).Err(err).Msg("Access denied for account")
-					continue
-				}
+			initLock.Lock()
+			for _, details := range svcsDetail {
+				client.ServicesManager.InitServices(details)
 			}
-			if errors.Is(err, errRetrievingCredentials) {
-				logger.Warn().Str("account", account.AccountName).Err(err).Msg("Could not retrieve credentials for account")
-				continue
-			}
-
-			return nil, err
-		}
-		account.Regions = findEnabledRegions(ctx, logger, account.AccountName, ec2.NewFromConfig(awsCfg), localRegions, account.DefaultRegion)
-		if len(account.Regions) == 0 {
-			logger.Warn().Str("account", account.AccountName).Err(err).Msg("No enabled regions provided in config for account")
-			continue
-		}
-		awsCfg.Region = account.Regions[0]
-		output, err := getAccountId(ctx, awsCfg)
-		if err != nil {
-			return nil, err
-		}
-		iamArn, err := arn.Parse(*output.Arn)
-		if err != nil {
-			return nil, err
-		}
-
-		for _, region := range account.Regions {
-			client.ServicesManager.InitServicesForPartitionAccountAndRegion(iamArn.Partition, *output.Account, region, initServices(region, awsCfg))
-		}
-		client.ServicesManager.InitServicesForPartitionAccountAndScope(iamArn.Partition, *output.Account, initServices(cloudfrontScopeRegion, awsCfg))
-	}
-	if len(client.ServicesManager.services) == 0 {
-		return nil, fmt.Errorf("no enabled accounts instantiated")
-	}
-	return &client, nil
-}
-
-func findEnabledRegions(ctx context.Context, logger zerolog.Logger, accountName string, ec2Client services.Ec2Client, localRegions []string, accountDefaultRegion string) []string {
-	// By default we should use the default region (us-east-1)
-	regionsToCheck := []string{defaultRegion}
-	// If user specifies a Default Region we should use it
-	if accountDefaultRegion != "" {
-		regionsToCheck = []string{accountDefaultRegion}
-		// If no default region and * is not specified we should use all specified regions
-	} else if len(localRegions) > 0 && !isAllRegions(localRegions) {
-		regionsToCheck = localRegions
-	}
-
-	for _, region := range regionsToCheck {
-		enabledRegions, err := getEnabledRegions(ctx, ec2Client, region)
-		if err != nil {
-			logger.Warn().Str("account", accountName).Err(err).Msgf("Failed to find disabled regions for account when checking: %s", region)
-			continue
-		}
-		filteredRegions := filterDisabledRegions(localRegions, enabledRegions)
-		if len(filteredRegions) > 0 {
-			return filteredRegions
-		}
-	}
-	return []string{}
-}
-
-func getEnabledRegions(ctx context.Context, ec2Client services.Ec2Client, region string) ([]types.Region, error) {
-	res, err := ec2Client.DescribeRegions(ctx,
-		&ec2.DescribeRegionsInput{AllRegions: aws.Bool(false)},
-		func(o *ec2.Options) {
-			o.Region = region
+			defer initLock.Unlock()
```

Comment:
Would move `defer` right after the `Lock()`, so that if something panics the lock is still intact (could prevent hanging)

Reply:
So this is a really weird diff that github created:
This is the actual code
```
		errorGroup.Go(func() error {
			svcsDetail, err := client.setupAWSAccount(gtx, logger, &awsPluginSpec, adminAccountSts, account)
			if err != nil {
				return err
			}
			initLock.Lock()
			for _, details := range svcsDetail {
				client.ServicesManager.InitServices(details)
			}
			defer initLock.Unlock()
			return nil
		})
```

Reply:
but updated it to:
```
			initLock.Lock()
			defer initLock.Unlock()
			for _, details := range svcsDetail {
				client.ServicesManager.InitServices(details)
			}
```

Reply:
Noticed the weird diff but still managed to parse it :) thanks for the update